File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        boolean[] v = new boolean[n];
        
        ArrayDeque<Integer> que = new ArrayDeque<>();
        for (int i=0; i<n; i++) {
            if (v[i]) continue;
            v[i] = true;
            que.add(i);
            answer++;
            while(!que.isEmpty()) {
                int now = que.poll();
                
                for (int j=0; j<n; j++) {
                    if (computers[now][j] == 0 || v[j]) continue;
                    v[j] = true;
                    que.add(j);
                }
                
            }
            
        }
        return answer;
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    que = deque()
    
    v = [False] * n
    for i in range(n):
        if v[i]: 
            continue
        
        v[i] = True
        answer += 1
        que.append(i)
        
        while que:
            now = que.popleft()
            for j in range(n):
                if now==j: 
                    continue
                if v[j] or computers[now][j]==0: 
                    continue
                
                v[j] = True
                que.append(j)
                
    
    
    return answer